homeassistant/components/zwave_js/services.py

```diff
@@ -497,23 +544,42 @@ async def async_set_config_parameter(self, service: ServiceCall) -> None:
                     property_key=property_key,
                     endpoint=endpoint,
                 )
+                if value_size is None
+                else node.endpoints[endpoint].async_set_raw_config_parameter_value(
+                    new_value,
+                    property_or_property_name,
+                    property_key=property_key,
+                    value_size=value_size,
+                    value_format=value_format,
+                )
                 for node in nodes
             ),
             return_exceptions=True,
         )
-        nodes_list = list(nodes)
-        for node, result in get_valid_responses_from_results(nodes_list, results):
-            zwave_value = result[0]
-            cmd_status = result[1]
-            if cmd_status == CommandStatus.ACCEPTED:
-                msg = "Set configuration parameter %s on Node %s with value %s"
-            else:
-                msg = (
-                    "Added command to queue to set configuration parameter %s on Node "
-                    "%s with value %s. Parameter will be set when the device wakes up"
-                )
-            _LOGGER.info(msg, zwave_value, node, new_value)
-        raise_exceptions_from_results(nodes_list, results)
+
+        def process_results(
+            nodes_or_endpoints_list: list[T], _results: list[Any]
+        ) -> None:
+            """Process results for given nodes or endpoints."""
+            for node_or_endpoint, result in get_valid_responses_from_results(
+                nodes_or_endpoints_list, _results
+            ):
+                zwave_value = result[0]
+                cmd_status = result[1]
+                if cmd_status == CommandStatus.ACCEPTED:
+                    msg = "Set configuration parameter %s on Node %s with value %s"
```

Comment:
This is existing code, but it doesn't seem covered by a test. Would be good to follow up with that.

Reply:
better to do it here because I found a bug!